packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal.sol

```diff
@@ -1,17 +1,17 @@
 // SPDX-License-Identifier: MIT
-pragma solidity 0.8.9;
+pragma solidity 0.8.24;
```

Comment:
If I am not mistaken, the reentrancy guard only needs to be applied to the message portal.

Do we need to upgrade the compiler for all the contracts in the code base?

Reply:
we were earlier using the specific version `0.8.9`,  but with `0.8.24` version, we had a choice that, either we upgrade the compiler versions for all contracts to `0.8.24`, or we use the [caret version](https://medium.com/@nareshmmr/why-pragma-solidity-0-x-x-is-important-in-solidity-smart-contracts-9fa669f0110f) which is `^0.8.9` in all contracts so it is compatible with `0.8.24` 

I chose to go with using the specific version

Reply:
Can you not configure several compilers in the hardhat config? There is a compilers section that accepts an array

Reply:
yeah, I was just trying to avoid using multiple compiler versions, but yeah will do that